client/widgets/MiscWidgets.h

```diff
@@ -164,17 +164,21 @@ class CMinorResDataBar : public CIntObject
 	~CMinorResDataBar();
 };
 
-/// Opens hero window by left-clicking on it
+/// Performs an action by left-clicking on it. Opens hero window by default
 class CHeroArea: public CIntObject
 {
-	const CGHeroInstance * hero;
-	std::shared_ptr<CAnimImage> portrait;
-
 public:
-	CHeroArea(int x, int y, const CGHeroInstance * _hero);
+	using ClickHandler = std::function<void()>;
```

Comment:
We usually use word "Handler" for subsystem names, functors are usually called callbacks or well, functors.
```suggestion
	using ClickFunctor = std::function<void()>;
```